### ISSUE TITLE
setRtpEncodingParameters() documentation/implementation/typings issue

### ISSUE THREAD

**Issue by nazar-pc** [versatica/mediasoup-client#123]
There is currently a problem with `producer.setRtpEncodingParameters()` method:
1) Both actual implementation in browser handlers and documentation expect `RTCRtpEncodingParameters` as an argument
2) Producer's API has TypeScript type defined as `RTCRtpEncodingParameters[]` instead (I guess it only compiles because there is a type information erasing in handlers with `params: any`)

https://github.com/versatica/mediasoup-client/blob/3aa7f7c3eca4ef39f5e8b3cdb448dbe5008cc71c/src/Producer.ts#L396-L398
https://github.com/versatica/mediasoup-client/blob/3aa7f7c3eca4ef39f5e8b3cdb448dbe5008cc71c/src/handlers/Chrome74.ts#L520-L541

So in TypeScript project it is possible to either specify something that is `RTCRtpEncodingParameters[]` and will not work or you have to cast it to `any` to be able to use `RTCRtpEncodingParameters` directly.

The more annoying thing is that `RTCRtpEncodingParameters` alone, which is the de-facto current implementation, doesn't allow applying different values to different encodings (simulcast).

I'd suggest is to add support for `RTCRtpEncodingParameters[]` such that it is possible to have different parameters for different encodings, but until next major version bump also support old plain `RTCRtpEncodingParameters` argument for those who rely on it.

If that sounds good, I can prepare a PR.

**Comment by ArmorDarks** [comment]
It seems that it should be rather that:

```diff
- parameters.encodings[idx] = { ...encoding, ...params };
+ parameters.encodings[idx] = { ...encoding, ...params[idx] };
```

**Comment by ibc** [comment]
It's a bug, it should be `RTCRtpEncodingParameters` (no array). Will fix it when I am back to real life.

**Comment by nazar-pc** [comment]
What about ability to set multiple encodings in case of simulcast then?

**Comment by ibc** [comment]
This method should just be removed, you can already get get the RtpSender. No need for sugar APIs. But it's too late to remove it.

BTW you can not change the number of encodings in runtime.

**Comment by ArmorDarks** [comment]
> This method should just be removed, you can already get get the RtpSender. No need for sugar APIs. But it's too late to remove it.

Yeap works fine. The only downside is that `producer.rtpParameters` are not updated properly in that case. Actual values can be obtained from the `producer.rtpSender.getParameters().encodings`. Though, they have a bit different interface signature.

> BTW you can not change the number of encodings in runtime.

Sure. But it's possible to change the values (let's say, `maxBitrate`) of each encoding. Current `setRtpEncodingParameters` is impractical since it will set the same value to all encodings

**Comment by ibc** [comment]
> Yeap works fine. The only downside is that `producer.rtpParameters` are not updated properly in that case. Actual values can be obtained from the `producer.rtpSender.getParameters().encodings`. Though, they have a bit different interface signature.

Right. `producer.rtpParameters` correspond to the original RTP parameters sent to mediasoup, not the later updated ones.
 
> > BTW you can not change the number of encodings in runtime.
> 
> Sure. But it's possible to change the values (let's say, `maxBitrate`) of each encoding. Current `setRtpEncodingParameters` is impractical since it will set the same value to all encodings

Indeed, that's why I say that this method should not exist.

**Comment by ibc** [comment]
Fixed in 3.6.12.
